api/api/management/commands/backfillmoderationdecision.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+
+from django.contrib.auth import get_user_model
+
+from django_tqdm import BaseCommand
+
+from api.models import AudioDecision, AudioReport, ImageDecision, ImageReport
+from api.models.media import DMCA, MATURE_FILTERED, NO_ACTION, PENDING
+
+
+class Command(BaseCommand):
+    help = "Back-fill the moderation decision table for a given media type."
+
+    @staticmethod
+    def add_arguments(parser):
+        parser.add_argument(
+            "--dry-run",
+            help="Count reports to process, and don't do anything else.",
+            type=bool,
+            default=True,
+            action=argparse.BooleanOptionalAction,
+        )
+        parser.add_argument(
+            "--media-type",
+            help="The media type to back-fill moderation decisions.",
+            type=str,
+            default="image",
+            choices=["image", "audio"],
+        )
+        parser.add_argument(
+            "--moderator",
+            help="The username of the moderator to attribute the decisions to.",
+            type=str,
+            default="opener",
+        )
+
+    def handle(self, *args, **options):
+        dry = options["dry_run"]
+        username = options["moderator"]
+        media_type = options["media_type"]
+
+        MediaReport = ImageReport
+        MediaDecision = ImageDecision
+        if media_type == "audio":
+            MediaReport = AudioReport
+            MediaDecision = AudioDecision
+
+        non_pending_reports = MediaReport.objects.filter(decision=None).exclude(
+            status=PENDING
+        )
+        count_to_process = non_pending_reports.count()
+
+        if dry:
+            self.info(
+                f"{count_to_process} {media_type} reports to back-fill. "
+                f"This is a dry run, exiting without making changes."
+            )
+            return
+
+        if not count_to_process:
+            self.info("No reports to process.")
+            return
+
+        t = self.tqdm(total=count_to_process)
+
+        User = get_user_model()
+        try:
+            moderator = User.objects.get(username=username)
+        except User.DoesNotExist:
+            t.error(f"User '{username}' not found.")
+            return
+
+        for report in non_pending_reports:
```

Comment:
This will pull every single non pending report into memory, as starting the iteration will evaluate the query set.

How many of these exist in production? If it's anything over ~100 I think we should chunk this.

Regardless of whether we chunk it up, we should use [`bulk_create`](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#bulk-create) rather than `.save()` in each Python loop iteration which creates `N` queries. It will return the decision models in order of iteration, so we can reliably zip them with the chunk of reports to set the decision on each report. Postgres supports returning the keys so the returned models will have PKs.

Reply:
Good suggestion! There are _only_ 135 non-pending reports, but it's a good practice to make these things more performance. I'll change it to do a bulk_create.

```sql
deploy@localhost:openledger> SELECT COUNT(*) FROM nsfw_reports WHERE status <> 'pending_review';
+-------+
| count |
|-------|
| 135   |
+-------+
SELECT 1
Time: 0.079s
```

Reply:
Done in [4d97b12](https://github.com/WordPress/openverse/pull/4415/commits/4d97b1223c42f1ea0096f63480dbfdfb448091a8).